core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -0,0 +1,137 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.model.Frame;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.ToString;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.rnorth.ducttape.TimeoutException;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.output.FrameConsumerResultCallback;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * 'Sidekick container' with the sole purpose of recording the VNC screen output from another container.
+ *
+ */
+@Getter
+@ToString
+public class VncRecordingContainer extends GenericContainer<VncRecordingContainer> {
+
+    private static final String RECORDING_FILE_NAME = "/screen.flv";
+
+    public static final String DEFAULT_VNC_PASSWORD = "secret";
+
+    public static final int DEFAULT_VNC_PORT = 5900;
+
+    private final String targetNetworkAlias;
+
+    private String vncPassword = DEFAULT_VNC_PASSWORD;
+
+    private int vncPort = 5900;
+
+    private int frameRate = 30;
+
+    public VncRecordingContainer(@NonNull GenericContainer<?> targetContainer) {
+        this(
+                targetContainer.getNetwork(),
+                targetContainer.getNetworkAliases().stream()
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("Target container must have a network alias"))
```

Comment:
Maybe we could make this less likely to happen by _always_ creating a default (random?) alias for a container whenever you join it to a network?

Does this already happen / would it cause problems if we did it?

Reply:
if container already started, we can't modify its aliases I think

Reply:
actually, we CAN alias during `connect`:
https://docs.docker.com/engine/reference/commandline/network_connect/#description

I'll check if it's possible with docker-java, will make users' live much easier